Correct output for dashboard filter variables

ericnormand opened the pull request:
Closes https://github.com/metabase/metabase/issues/39385

### Description

The dashboard text previews were incomplete. I've completed them for various cases:

`Before DATE`
`After DATE`
`On DATE`
`Exclude {hours|days|quarters|months}`

### How to verify

1. Create a dashboad.
2. Add a filter type Date Picker called Date
3. Add a text box. Add the text {{Date}}
4. Save it.
5. Click edit. Click the date picker filter.
6. In the text box, select Date as the variable to map to.
7. Click save.
8. Select a date: Specific Date-> Before-> Pick a date
9. Observe the text "Before ..." in the text box.

ericnormand committed:
Correct output for dashboard filter variables

ericnormand committed:
Whitespace

bshepherdson commented on src/metabase/models/params/shared.cljc:
```diff
@@ -82,15 +82,87 @@
+#?(:clj (def ^:private day-out (DateTimeFormatter/ofPattern "EEEE")))
+
+(defn- format-day [value locale]
+  #?(:cljs (let [t (.locale moment)
+                 _ (.locale moment "en") ;; always read in en locale
+                 s (-> value
+                       (moment "ddd")
+                       (.locale locale)
+                       (.format "dddd"))]
+             (.locale moment t)
+             s)
```
It took me a second to figure this out. It really needs a better factoring, since this is pretty janky and it's not at all obvious what's going on at first.

There's `u.time/string->timestamp` as well, but I don't know if it's quite as flexible as this. We could add a `:locale` to its `options` map likewise.

ericnormand committed:
Add node script for testing clojurescript

ericnormand committed:
Enhance format-unit and add parse-unit.

parse-unit is new.

format-unit now works with moment (cljs) or java.time (clj) values, not just strings and numbers. I also added two new
units (:day-of-week-abbrev and :month-of-year-full).

ericnormand committed:
Refactor to use time/parse-unit and time/format-unit

bshepherdson commented on src/metabase/util/time/impl.cljs:
```diff
@@ -262,17 +296,48 @@
+   (when-some [format (get unit-formats unit)]
+     (moment input format)))
+  ([input unit locale]
+   (let [temp (.locale moment)]     ;; 1. save current locale
+     (try
+       (.locale moment locale)      ;; 2. set new locale for subsequent parse
+       (parse-unit input unit)
+       (finally
+         (.locale moment temp)))))) ;; 3. set locale to original
```
`temp` is misleading here. It's actually the opposite - we're saving the `original` locale, setting it to `locale` temporarily, and then putting the `original` back.  I think `saved` or `original` is the right name.
